deepinv/tests/test_models.py

```diff
@@ -25,31 +25,46 @@ def imsize_1_channel():
 
 
 model_list = [
-    "unet",
+    "autoencoder",
+    "bm3d",
     "drunet",
-    "scunet",
     "dncnn",
+    "gsdrunet",
+    "median",
+    "scunet",
+    "swinir",
+    "tgv",
+    "tv",
+    "unet",
     "waveletprior",
     "waveletdict",
     "waveletdict_hard",
     "waveletdict_topk",
-    "tgv",
-    "tv",
-    "median",
-    "autoencoder",
-    "gsdrunet",
-    "swinir",
 ]
 
-try:  # install of BM3D may fail on some architectures (arm64)
-    from dinv.models.bm3d import bm3d
 
-    model_list.append("bm3d")
-except ImportError:
-    print("Could not find bm3d; not testing bm3d.")
+def choose_denoiser(name, imsize):
 
+    if name in ("waveletprior", "waveletdict"):
+        pytest.importorskip(
+            "pytorch_wavelets",
+            reason="This test requires pytorch_wavelets. It should be "
+            "installed with `pip install "
+            "git+https://github.com/fbcotter/pytorch_wavelets.git`"
+        )
+    if name == "bm3d":
+        pytest.importorskip(
+            "bm3d",
+            reason="This test requires bm3d. It should be "
+            "installed with `pip install bm3d`"
+        )
+    if name == "swinir":
```

Comment:
```
if name in ["swinir", "scunet"]:
```

Reply:
also: would be clean to add a test when additional dependencies are not installed (pytorch_wavelets, timm, bm3d...). Would follow a similar pattern as here